src/logic/wallets/onboard.ts

```diff
@@ -51,11 +53,13 @@ const getOnboardConfiguration = () => {
   }
 }
 
-let currentOnboardInstance = Onboard(getOnboardConfiguration())
-const onboard = (): API => {
-  if (!currentOnboardInstance) {
+let currentOnboardInstance
+export const onboard = (): API => {
+  const chainId = getNetworkId()
+  if (!currentOnboardInstance || currentOnboardInstance.getState().appNetworkId !== parseInt(chainId, 10)) {
```

Comment:
I suggest we stringily `appNetworkId` instead of parsing `chainId`

Reply:
Yes, is more aligned to our app that should handle it as a string :+1: